Replaced `TimeSpan with `std::chrono::duration`

mandrek44 opened the pull request:
I replaced implementation of `TimeSpan` with `std::chrono::duration<uint64_t>`. 
Some methods were removed because they duplicated `std::chrono::duration` built-in functionality.
I left `TimePoint` becasue it seems tailored for this system needs.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pw-sat2/pwsat2obc/66)
<!-- Reviewable:end -->

Novakov commented:
Reviewed 14 of 14 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zQs1EXa251LlAaboi:-K_zQs1EXa251LlAaboj:b-l9739l) (no related file):*
There are also few functions accepting time in `base/os.h`. It would be nice to have them changed to `std::chrono` as well.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66)*
<!-- Sent from Reviewable.io -->

zaq32 commented:
Review status: all files reviewed at latest revision, 12 unresolved discussions.

---

*[libs/time/TimePoint.cpp, line 5 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zfIEISzclmoh-SA9M:-K_zfIEISzclmoh-SA9N:b13m3xu) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/TimePoint.cpp#L5)):*
> ```cpp
> TimeSpan TimeSpanFromMilliseconds(uint64_t milliseconds)
> {
>     const TimeSpan span(milliseconds);
> ```

this would be much safer (in case anyone will ever get the idea of changing the base time unit):
```return std::chrono::duration_cast<TimeSpan>(std::chrono::milliseconds(milliseconds));```
```


---

*[libs/time/TimePoint.cpp, line 11 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zeY-Vdl5odBHM7Vsg:-K_zeY-Vdl5odBHM7Vsh:b-z379ok) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/TimePoint.cpp#L11)):*
> ```cpp
> TimeSpan TimeSpanFromSeconds(uint32_t seconds)
> {
>     return TimeSpanFromMilliseconds(seconds * 1000ull);
> ```

consider this:
```return std::chrono::duration_cast<TimeSpan>(std::chrono::seconds(seconds));```
```


---

*[libs/time/TimePoint.cpp, line 16 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zexzSBkmBNoXmqzmn:-K_zexzSBkmBNoXmqzmo:b-dyw58k) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/TimePoint.cpp#L16)):*
> ```cpp
> TimeSpan TimeSpanFromMinutes(uint32_t minutes)
> {
>     return TimeSpanFromMilliseconds(minutes * 60000ull);
> ```

consider this:
```return std::chrono::duration_cast<TimeSpan>(std::chrono::minutes(minutes));```
```


---

*[libs/time/TimePoint.cpp, line 21 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zf-fRLmFjagjXwC_O:-K_zf-fRLmFjagjXwC_P:b-ap94wi) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/TimePoint.cpp#L21)):*
> ```cpp
> TimeSpan TimeSpanFromHours(uint32_t hours)
> {
>     return TimeSpanFromMilliseconds(hours * 3600000ull);
> ```

consider this:
```return std::chrono::duration_cast<TimeSpan>(std::chrono::hours(hours));```
```


---

*[libs/time/TimePoint.cpp, line 26 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zf4pfYfEZSDZnMbqy:-K_zf4pfYfEZSDZnMbqz:b-2km9vk) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/TimePoint.cpp#L26)):*
> ```cpp
> TimeSpan TimeSpanFromDays(uint32_t days)
> {
>     return TimeSpanFromMilliseconds(days * 24ull * 3600000ull);
> ```

consider this:
```return std::chrono::duration_cast<TimeSpan>(std::chrono::hours(days * 24));```
```


---

*[libs/time/TimePoint.cpp, line 31 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zbjgAjUPjRmFaIstm:-K_zbjgAjUPjRmFaIstn:b-snl7sk) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/TimePoint.cpp#L31)):*
> ```cpp
> uint32_t TimeSpanToSeconds(TimeSpan span)
> {
>     return span.count() / 1000;
> ```

for time unit conversion use std::chrono::duration_cast.

---

*[libs/time/TimePoint.cpp, line 53 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zbwBANuNUP4nF_E6g:-K_zbwBANuNUP4nF_E6h:b558eh3) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/TimePoint.cpp#L53)):*
> ```cpp
> {
>     TimePoint point = {};
>     uint64_t span = timeSpan.count();
> ```

this code assumes that time spans are in ms. Now we can lift this assumption with std::chrono::duration_cast

---

*[libs/time/include/time/TimePoint.h, line 41 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zcPrdSGYKYb912KUE:-K_zcPrdSGYKYb912KUF:b-vzq8sj) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/include/time/TimePoint.h#L41)):*
> ```objc
>  * beginning of the mission itself.
>  */
> typedef std::chrono::duration<uint64_t, std::milli> TimeSpan;
> ```

There is already standard type for durations in ms: std::chrono::milliseconds. Is there any reason for not using that type?

---

*[src/commands/time.cpp, line 18 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zdyaseSOrnEv2QpP-:-K_zdyaseSOrnEv2QpP0:br6ih9o) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/src/commands/time.cpp#L18)):*
> ```cpp
>     char* tail;
>     const TimeSpan targetTime = TimeSpanFromSeconds(strtoul(argv[0], &tail, 10));
>     LOGF(LOG_LEVEL_INFO, "Jumping to time %lu\n", static_cast<std::uint32_t>(targetTime.count() / 1000));
> ```

use std::chrono::duration_cast.

---

*[src/commands/time.cpp, line 32 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_ze-zEovMtbcKWkGp5:-K_ze-zEovMtbcKWkGp6:b-41d6kq) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/src/commands/time.cpp#L32)):*
> ```cpp
>     char* tail;
>     const TimeSpan targetTime = TimeSpanFromMilliseconds(strtoul(argv[0], &tail, 10));
>     LOGF(LOG_LEVEL_INFO, "Advancing time by '%lu' seconds\n", static_cast<std::uint32_t>(targetTime.count() / 1000));
> ```

 use std::chrono::duration_cast.

---

*[src/commands/time.cpp, line 41 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_ze0vqRb-YmFXAEjZO:-K_ze0vqRb-YmFXAEjZP:b-41d6kq) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/src/commands/time.cpp#L41)):*
> ```cpp
>     UNREFERENCED_PARAMETER(argv);
>     TimeSpan span = Main.timeProvider.GetCurrentTime().Value;
>     Main.terminal.Printf("%lu", static_cast<std::uint32_t>(span.count() / 1000));
> ```

 use std::chrono::duration_cast.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66)*
<!-- Sent from Reviewable.io -->

Novakov commented:
![](https://github.trello.services/images/mini-trello-icon.png) [STD::Chrono [C++]](https://trello.com/c/LK25zXP2/94-std-chrono-c)

mandrek44 commented:
Review status: all files reviewed at latest revision, 12 unresolved discussions, some commit checks failed.

---

*[libs/time/include/time/TimePoint.h, line 41 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zcPrdSGYKYb912KUE:-Ka3SbFVCvN3TL4mJRE8:b-ot4sby) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/include/time/TimePoint.h#L41)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary><blockquote>

There is already standard type for durations in ms: std::chrono::milliseconds. Is there any reason for not using that type?
</blockquote></details>

`std::chrono::milliseconds` uses `int64_t`, we were using `uint64_t` - I didn't want to change the underlying type, but If you think it's ok to use `int64_t` then I'll change it

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66)*
<!-- Sent from Reviewable.io -->

zaq32 commented:
Review status: all files reviewed at latest revision, 12 unresolved discussions, some commit checks failed.

---

*[libs/time/include/time/TimePoint.h, line 41 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zcPrdSGYKYb912KUE:-Ka8nxRXP-Pc2C0rjiVl:bl7su2i) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/include/time/TimePoint.h#L41)):*
<details><summary><i>Previously, mandrek44 (Mandro) wrote…</i></summary><blockquote>

`std::chrono::milliseconds` uses `int64_t`, we were using `uint64_t` - I didn't want to change the underlying type, but If you think it's ok to use `int64_t` then I'll change it
</blockquote></details>

This will not be a problem for us.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66)*
<!-- Sent from Reviewable.io -->

mandrek44 commented:
Review status: 2 of 52 files reviewed at latest revision, 12 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zQs1EXa251LlAaboi:-KaIkVFcy1P4L3W2IAnf:b6zhsqm) (no related file):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

There are also few functions accepting time in `base/os.h`. It would be nice to have them changed to `std::chrono` as well.
</blockquote></details>

I removed the `OSTaskTimeSpan` and replaced with `std::chrono::duration`

---

*[libs/time/TimePoint.cpp, line 5 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zfIEISzclmoh-SA9M:-KaIjwA5S7wrmIhd-Kg5:b-phc23a) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/TimePoint.cpp#L5)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary><blockquote>

this would be much safer (in case anyone will ever get the idea of changing the base time unit):
```return std::chrono::duration_cast<TimeSpan>(std::chrono::milliseconds(milliseconds));```
</blockquote></details>

I removed `TimeSpan` completely and replaced it directly with `std::chrono::duration` - methods like this are not needed any more

---

*[libs/time/TimePoint.cpp, line 11 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zeY-Vdl5odBHM7Vsg:-KaIkDWgPn8MkJgmj-ku:b-5u845b) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/TimePoint.cpp#L11)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary><blockquote>

consider this:
```return std::chrono::duration_cast<TimeSpan>(std::chrono::seconds(seconds));```

</blockquote></details>

I removed `TimeSpan` completely

---

*[libs/time/TimePoint.cpp, line 16 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zexzSBkmBNoXmqzmn:-KaIkDsWhsVdo_poMG2F:b-5u845b) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/TimePoint.cpp#L16)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary><blockquote>

consider this:
```return std::chrono::duration_cast<TimeSpan>(std::chrono::minutes(minutes));```

</blockquote></details>

I removed `TimeSpan` completely

---

*[libs/time/TimePoint.cpp, line 21 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zf-fRLmFjagjXwC_O:-KaIkESJrl9mucc6iPKO:b-5u845b) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/TimePoint.cpp#L21)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary><blockquote>

consider this:
```return std::chrono::duration_cast<TimeSpan>(std::chrono::hours(hours));```
</blockquote></details>

I removed `TimeSpan` completely

---

*[libs/time/TimePoint.cpp, line 26 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zf4pfYfEZSDZnMbqy:-KaIkEylheivxsKc-Y8w:b-5u845b) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/TimePoint.cpp#L26)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary><blockquote>

consider this:
```return std::chrono::duration_cast<TimeSpan>(std::chrono::hours(days * 24));```
</blockquote></details>

I removed `TimeSpan` completely

---

*[libs/time/TimePoint.cpp, line 31 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zbjgAjUPjRmFaIstm:-KaIkGHUs470DRnUN9e-:b-5u845b) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/TimePoint.cpp#L31)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary><blockquote>

for time unit conversion use std::chrono::duration_cast.
</blockquote></details>

I removed `TimeSpan` completely

---

*[libs/time/TimePoint.cpp, line 53 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zbwBANuNUP4nF_E6g:-KaIkI3FaBpz9WBfXJOG:b-5u845b) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/TimePoint.cpp#L53)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary><blockquote>

this code assumes that time spans are in ms. Now we can lift this assumption with std::chrono::duration_cast
</blockquote></details>

I removed `TimeSpan` completely

---

*[libs/time/include/time/TimePoint.h, line 41 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zcPrdSGYKYb912KUE:-KaIkKIE1kVx6PnKikvD:b32wtd) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/libs/time/include/time/TimePoint.h#L41)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary><blockquote>

This will not be a problem for us.
</blockquote></details>

ok

---

*[src/commands/time.cpp, line 18 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_zdyaseSOrnEv2QpP-:-KaIkNwvgCNRN06yugdT:b32wtd) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/src/commands/time.cpp#L18)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary><blockquote>

use std::chrono::duration_cast.
</blockquote></details>

ok

---

*[src/commands/time.cpp, line 32 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_ze-zEovMtbcKWkGp5:-KaIkRfInxcYbv1-1UnV:b32wtd) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/src/commands/time.cpp#L32)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary><blockquote>

 use std::chrono::duration_cast.
</blockquote></details>

ok

---

*[src/commands/time.cpp, line 41 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-K_ze0vqRb-YmFXAEjZO:-KaIkc0N2vF_-r5LwH5x:b32wtd) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/3bec947364e5c8e1c1bf5a8eb9f7aa8039a54874/src/commands/time.cpp#L41)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary><blockquote>

 use std::chrono::duration_cast.
</blockquote></details>

ok

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66)*
<!-- Sent from Reviewable.io -->

Novakov commented:
Reviewed 50 of 50 files at r3.
Review status: all files reviewed at latest revision, 26 unresolved discussions.

---

*[libs/adcs/adcs.cpp, line 39 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIlJm6YKxvHlFVs15M:-KaIlJm6YKxvHlFVs15N:b-r6rvb7) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/adcs/adcs.cpp#L39)):*
> ```cpp
> 
>         LOGF(LOG_LEVEL_TRACE, "[ADCS]Running ADCS loop. Mode: %d", context->CurrentMode);
>         System::SleepTask(std::chrono::seconds(5));
> ```

I prefer using literals (http://en.cppreference.com/w/cpp/chrono/operator%22%22s)

---

*[libs/drivers/i2c/efm.cpp, line 19 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaImZ5tdh_vuXu_s0VO:-KaImZ5tdh_vuXu_s0VP:b-edp1u7) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/drivers/i2c/efm.cpp#L19)):*
> ```cpp
> I2CResult I2CLowLevelBus::ExecuteTransfer(I2C_TransferSeq_TypeDef* seq)
> {
>     Lock lock(this->_lock, std::chrono::milliseconds(MAX_DELAY));
> ```

MAX_DELAY is a bit specific one as it defines inifinite timeout. It maybe better to define constexpr `InfiniteTimeout` and use it here

---

*[libs/drivers/n25q/Include/n25q/n25q.h, line 210 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIomt9aafv89_jpGnp:-KaIomtABNvb5NOMoxkR:bj6ss7q) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/drivers/n25q/Include/n25q/n25q.h#L210)):*
> ```objc
>             /** Erase chip operation timeout */
>             static constexpr std::chrono::milliseconds EraseChipTimeOut =
>                 std::chrono::milliseconds(static_cast<uint64_t>(1.2 * (250 * 1000)));
> ```

This expressions can be simplified as some of them are converting seconds to milliseconds by multiplying by 1000

---

*[libs/free_rtos_wrapper/os.cpp, line 21 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIpt2ip5jHwOGgFFL5:-KaIpt2ip5jHwOGgFFL6:bwx9keq) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/free_rtos_wrapper/os.cpp#L21)):*
> ```cpp
>     {
>         return portMAX_DELAY;
>     }
> ```

see remark about `InfiniteTimeout`

---

*[libs/free_rtos_wrapper/os.cpp, line 203 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIqGWxZhcqVheubEqF:-KaIqGWxZhcqVheubEqG:bv3vsf1) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/free_rtos_wrapper/os.cpp#L203)):*
> ```cpp
> std::chrono::milliseconds System::GetUptime()
> {
>     return std::chrono::milliseconds(static_cast<uint64_t>(portTICK_PERIOD_MS * xTaskGetTickCount()));
> ```

I think it's possible to define duration type for single RTOS tick and use duration casting to convert into into milliseconds

---

*[libs/mission/Include/mission/main.hpp, line 312 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIqY7JOI9B6ZeJyYIz:-KaIqY7JOI9B6ZeJyYJ-:b-cklbv2) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/mission/Include/mission/main.hpp#L312)):*
> ```cpp
>         {
>             System::EventGroupSetBits(this->eventGroup, PauseRequestFlag);
>             System::EventGroupWaitForBits(this->eventGroup, PauseAckFlag, false, true, std::chrono::milliseconds(MAX_DELAY));
> ```

See remark about `InfiniteTimeout`

---

*[libs/time/timer.cpp, line 72 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIrCyF0dhQE0kkj8c6:-KaIrCyF0dhQE0kkj8c7:b-cklbv2) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/time/timer.cpp#L72)):*
> ```cpp
> void TimeProvider::AdvanceTime(milliseconds delta)
> {
>     if (OS_RESULT_FAILED(System::TakeSemaphore(timerLock, milliseconds(MAX_DELAY))))
> ```

See remark about `InfiniteTimeout`

---

*[libs/time/timer.cpp, line 91 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIrEw-7dgy070SuWbc:-KaIrEw-7dgy070SuWbd:b-cklbv2) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/time/timer.cpp#L91)):*
> ```cpp
> {
>     const milliseconds span = TimePointToTimeSpan(pointInTime);
>     if (OS_RESULT_FAILED(System::TakeSemaphore(timerLock, milliseconds(MAX_DELAY))))
> ```

See remark about `InfiniteTimeout`

---

*[libs/time/timer.cpp, line 109 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIrGIoC4yU_KYmSZKp:-KaIrGIoC4yU_KYmSZKq:b-cklbv2) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/time/timer.cpp#L109)):*
> ```cpp
> Option<milliseconds> TimeProvider::GetCurrentTime()
> {
>     if (OS_RESULT_FAILED(System::TakeSemaphore(timerLock, milliseconds(MAX_DELAY))))
> ```

See remark about `InfiniteTimeout`

---

*[libs/time/timer.cpp, line 134 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIrHQFWWQB6qnfFYMR:-KaIrHQFWWQB6qnfFYMS:b-cklbv2) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/time/timer.cpp#L134)):*
> ```cpp
> void TimeProvider::ProcessChange(TimerState state)
> {
>     if (OS_RESULT_FAILED(System::TakeSemaphore(notificationLock, milliseconds(MAX_DELAY))))
> ```

See remark about `InfiniteTimeout`

---

*[libs/time/timer.cpp, line 269 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIrdm799RjGUIZPvX4:-KaIrdm799RjGUIZPvX5:by57bvj) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/time/timer.cpp#L269)):*
> ```cpp
>     }
> 
>     uint8_t buffer[sizeof(milliseconds)];
> ```

That's a bit risky. You are using size of duration object not it's underlying type. Use `milliseconds::rep`

---

*[libs/time/timer.cpp, line 329 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIrQkaZMBV_0LPd4ek:-KaIrQkaZMBV_0LPd4el:b-cklbv2) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/time/timer.cpp#L329)):*
> ```cpp
>         }
> 
>         if (OS_RESULT_FAILED(System::PulseWait(TickNotification, milliseconds(MAX_DELAY))))
> ```

See remark about `InfiniteTimeout`

---

*[libs/time/include/time/TimePoint.h, line 60 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIqyKgTflokJRzOG0_:-KaIqyKhV1oK3EmJy-ex:bph1757) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/time/include/time/TimePoint.h#L60)):*
> ```objc
>  * @return Decoded time point object.
>  */
> TimePoint TimePointFromTimeSpan(std::chrono::milliseconds span);
> ```

It will be better to name it `TimePointFromDuration`

---

*[libs/yaffs_glue/yaffs_os_glue.cpp, line 22 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIrxujq9OJpggS0AgZ:-KaIrxujq9OJpggS0Ag_:b-cklbv2) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/yaffs_glue/yaffs_os_glue.cpp#L22)):*
> ```cpp
> void yaffsfs_Lock(void)
> {
>     System::TakeSemaphore(yaffsLock, std::chrono::milliseconds(MAX_DELAY));
> ```

See remark about `InfiniteTimeout`

---

*[unit_tests/MissionPlan/MissionTestHelpers.h, line 30 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIsXIKLf-fbti2brUc:-KaIsXIKLf-fbti2brUd:bpgocq5) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/unit_tests/MissionPlan/MissionTestHelpers.h#L30)):*
> ```objc
>                 milisecond);
>         }
>     };
> ```

Didn't tested that but rapidcheck seems to support `std::chrono::duration` of out box

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66)*
<!-- Sent from Reviewable.io -->

mandrek44 commented:
Review status: 17 of 52 files reviewed at latest revision, 13 unresolved discussions.

---

*[libs/drivers/i2c/efm.cpp, line 19 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaImZ5tdh_vuXu_s0VO:-KaOvmUglc_f50LmEFQZ:b-896fix) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/drivers/i2c/efm.cpp#L19)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

MAX_DELAY is a bit specific one as it defines inifinite timeout. It maybe better to define constexpr `InfiniteTimeout` and use it here
</blockquote></details>

Done.

---

*[libs/drivers/n25q/Include/n25q/n25q.h, line 210 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIomt9aafv89_jpGnp:-KaOw1GO85TlrM17wGjQ:b-vxj1yg) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/drivers/n25q/Include/n25q/n25q.h#L210)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

This expressions can be simplified as some of them are converting seconds to milliseconds by multiplying by 1000
</blockquote></details>

I left the `1.2 * (0.8 * 1000)` but changed the others

---

*[libs/free_rtos_wrapper/os.cpp, line 21 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIpt2ip5jHwOGgFFL5:-KaOwAEWVYGUcodYIWsC:b-896fix) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/free_rtos_wrapper/os.cpp#L21)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

see remark about `InfiniteTimeout`
</blockquote></details>

Done.

---

*[libs/mission/Include/mission/main.hpp, line 312 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIqY7JOI9B6ZeJyYIz:-KaOwCJnDSK1NajDg2iv:b-896fix) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/mission/Include/mission/main.hpp#L312)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

See remark about `InfiniteTimeout`
</blockquote></details>

Done.

---

*[libs/time/timer.cpp, line 72 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIrCyF0dhQE0kkj8c6:-KaOwE8YscTS9Hzv2t6r:b-896fix) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/time/timer.cpp#L72)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

See remark about `InfiniteTimeout`
</blockquote></details>

Done.

---

*[libs/time/timer.cpp, line 91 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIrEw-7dgy070SuWbc:-KaOwF05V3ThnboBk2zH:b-896fix) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/time/timer.cpp#L91)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

See remark about `InfiniteTimeout`
</blockquote></details>

Done.

---

*[libs/time/timer.cpp, line 109 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIrGIoC4yU_KYmSZKp:-KaOwFp_YhsaRE_kXCgY:b-896fix) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/time/timer.cpp#L109)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

See remark about `InfiniteTimeout`
</blockquote></details>

Done.

---

*[libs/time/timer.cpp, line 134 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIrHQFWWQB6qnfFYMR:-KaOwGRr69Y9jKJAC5l2:b-896fix) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/time/timer.cpp#L134)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

See remark about `InfiniteTimeout`
</blockquote></details>

Done.

---

*[libs/time/timer.cpp, line 269 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIrdm799RjGUIZPvX4:-KaOwH9W7Z4gMMsBRKgc:b-896fix) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/time/timer.cpp#L269)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

That's a bit risky. You are using size of duration object not it's underlying type. Use `milliseconds::rep`
</blockquote></details>

Done.

---

*[libs/time/timer.cpp, line 329 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIrQkaZMBV_0LPd4ek:-KaOwHi1zJmo06RmU--d:b-896fix) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/time/timer.cpp#L329)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

See remark about `InfiniteTimeout`
</blockquote></details>

Done.

---

*[libs/time/include/time/TimePoint.h, line 60 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIqyKgTflokJRzOG0_:-KaOwKkuPT_SId2lQYCP:b-896fix) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/time/include/time/TimePoint.h#L60)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

It will be better to name it `TimePointFromDuration`
</blockquote></details>

Done.

---

*[libs/yaffs_glue/yaffs_os_glue.cpp, line 22 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIrxujq9OJpggS0AgZ:-KaOwLsHbyzyGSN0Kf9p:b-896fix) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/yaffs_glue/yaffs_os_glue.cpp#L22)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

See remark about `InfiniteTimeout`
</blockquote></details>

Done.

---

*[unit_tests/MissionPlan/MissionTestHelpers.h, line 30 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIsXIKLf-fbti2brUc:-KaOupIdt3Fk5usyVfF-:bvrd6d9) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/unit_tests/MissionPlan/MissionTestHelpers.h#L30)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

Didn't tested that but rapidcheck seems to support `std::chrono::duration` of out box
</blockquote></details>

It seems so, at least according to docs: https://github.com/emil-e/rapidcheck/blob/master/doc/generators.md

Do we want to use it in some specific test?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66)*
<!-- Sent from Reviewable.io -->

Novakov commented:
Reviewed 35 of 36 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[libs/drivers/n25q/Include/n25q/n25q.h, line 210 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIomt9aafv89_jpGnp:-KaOy3fTogyv0TmoqlCv:b-nmjabd) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/drivers/n25q/Include/n25q/n25q.h#L210)):*
<details><summary><i>Previously, mandrek44 (Mandro) wrote…</i></summary><blockquote>

I left the `1.2 * (0.8 * 1000)` but changed the others
</blockquote></details>

Why? It's simple 800ms + 20%

---

*[unit_tests/MissionPlan/MissionTestHelpers.h, line 30 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIsXIKLf-fbti2brUc:-KaOyrQ43UqyLDcIi7N1:b-upu73y) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/unit_tests/MissionPlan/MissionTestHelpers.h#L30)):*
<details><summary><i>Previously, mandrek44 (Mandro) wrote…</i></summary><blockquote>

It seems so, at least according to docs: https://github.com/emil-e/rapidcheck/blob/master/doc/generators.md

Do we want to use it in some specific test?
</blockquote></details>

System state related tests are using time value - that's why arbitrary was added. Let's try removing it and running all tests 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66)*
<!-- Sent from Reviewable.io -->

mandrek44 commented:
Review status: 50 of 52 files reviewed at latest revision, 2 unresolved discussions.

---

*[libs/drivers/n25q/Include/n25q/n25q.h, line 210 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIomt9aafv89_jpGnp:-KaSHLWD7UxvqXlZ4Myn:bb74x83) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/libs/drivers/n25q/Include/n25q/n25q.h#L210)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

Why? It's simple 800ms + 20%
</blockquote></details>

True

---

*[unit_tests/MissionPlan/MissionTestHelpers.h, line 30 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66#-KaIsXIKLf-fbti2brUc:-KaSHN45tNzQOK065Axa:b-1tmomh) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4cc9a5c7bdc7820ff1a54360948a627e65e1a35e/unit_tests/MissionPlan/MissionTestHelpers.h#L30)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary><blockquote>

System state related tests are using time value - that's why arbitrary was added. Let's try removing it and running all tests 
</blockquote></details>

I removed the generator - all seems to work ok

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66)*
<!-- Sent from Reviewable.io -->

Novakov commented:
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/66)*
<!-- Sent from Reviewable.io -->
